src/Microsoft.Identity.Client/ClientApplicationBase.cs

```diff
@@ -165,12 +166,15 @@ internal Authority GetAuthority(IAccount account)
                 customAuthority,
                 cache,
                 commonParameters,
-                CreateRequestContext());
+                requestContext);
         }
 
-        private RequestContext CreateRequestContext()
+        // This implementation should ONLY be called for cases where we aren't participating in
+        // MATS telemetry but still need a requestcontext/logger, such as "GetAccounts()".
+        // For service calls, the request context should be created in the **Executor classes as part of request execution.
+        private RequestContext CreateRequestContext(Guid telemetryCorrelationId)
```

Comment:
>telemetryCorrelationId [](start = 57, length = 22)

why would this not be the std. correlation id? is it required to have a special telemetryCorrelationId?

Reply:
I have the same question. Would be good to add comments to clarify.

Reply:
it doesn't have to be "special" but in the main execution calls, the telemetry correlation id is created WAY before we get into client application base (it's part of the API builders) so we need to keep it around since by the time we get into here we've probably already started the ApiEvent with the known telemetry id.  this code path (the builders) isn't hit when we call things like GetAccounts() so we still need to pass in a telemetry correlation id but it's created at the time of the call to GetAccounts instead of being passed in from higher up.

---
In reply to: [264539972](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/955#discussion_r264539972) [](ancestors = 264539972)